WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```
@@ -685,6 +686,11 @@ private extension MainTabBarController {
             cachePOSTabVisibility(siteID: siteID, isPOSTabVisible: isPOSTabVisible)
             updateTabViewControllers(isPOSTabVisible: isPOSTabVisible)
             viewModel.loadHubMenuTabBadge()
+
+            // Trigger POS catalog sync if tab is visible and feature flag is enabled
+            if isPOSTabVisible, ServiceLocator.featureFlagService.isFeatureFlagEnabled(.pointOfSaleLocalCatalogi1) {
+                await triggerPOSCatalogSyncIfNeeded(for: siteID)
```

nit: WDYT about moving `posCatalogSyncCoordinator` initialization here before this line to consolidate the catalog sync code? So that if the `updateTabViewControllers` call ever gets moved after this code by accident, it won't just silently skip the full sync.

---

❓ Have we decided to trigger the initial full sync if the store is **eligible for POS**, or if the **POS tab is shown** (not necessarily eligible)? If the former, we might need to check for eligibility from `posEligibilityChecker`. Though, it might be simpler to perform a full sync for stores with POS tab visible so that once they fix the ineligibility issue it enters POS right away. I'm not sure if we made a decision cross-platform. In any case, we can make any eligibility updates separately.

---

I'm not aware of a decision at that level of detail. I can see arguments for either. How quickly after the tab being shown do we usually know if the store's actually eligible?

If those are generally very close together, we can leave the sync until we know for sure. If it takes some time, e.g. they need to open the tab, I think it's better to sync preemptively.

---

> How quickly after the tab being shown do we usually know if the store's actually eligible?
>
> If those are generally very close together, we can leave the sync until we know for sure. If it takes some time, e.g. they need to open the tab, I think it's better to sync preemptively.

Currently, we only check the country eligibility & POS remote feature flag on site launch for the POS tab **visibility**. The rest of the POS eligibility (iOS version, currency, WC plugin version, feature switch in core) are checked when the user taps on the POS tab.

I think we can include the POS tab **visibility check** in the Local Catalog feature check. Asked a question in p1757465589133769/1757414787.869759-slack-C070SJRA8DP to sync with Android.